src/HOFs/propFormatter.js

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react'
+import { PARAMS_EXP_MAP } from "./constants"
+
+/**
+ * Creates a 1-step, or complete, URL from `domain` and `src` Strings.
+ *
+ * - First, the function checks if src has a defined `domain`. If it does, it 
+ * checks to see if `src` has a scheme, and prepends "http" or "https" as needed
+ * - Otherwise, formatSrc formats `domain` and `src` Strings.
+ *   - First it strips the two strings of the  leading and `/` or trailing `/` 
+ *     slash characters.
+ *   - Then, it joins the two strings on a `/` character. IE, 
+ *    `strippedDomain + "/" + strippedSrc`.
+ *   - If `domain` String argument `null` or `undefined`, the function returns
+ *    the original `src` String.
+ *
+ * @param {String} src - URL that is either 1-step or 2-step
+ * @param {String} domain - Domain string, optional
+ * @returns 1-step, or complete, URL String. Ex, _assets.ix.net/foo/bar.jpg_
+ */
+export function formatSrc(src, domain, useHTTPS = true) {
```

Comment:
Althought there is nothing wrong with this function at all, it does seem a bit silly to combine the path and domain into a full src just to be destructured later into `domain` and `path` again. I know the plan was not to change the internals of ReactImgix and I applaud this PR for doing that, but I wonder if it would be better just to rip the bandaid off and fix this problem properly. Another option is to wait until `@imgix/js-core` supports 1-step URLs and then just remove the destructuring behaviour inside ReactImgix then.

Ignore this if you have other plans already.